depthmapX/views/tableview/tableview.cpp

```diff
@@ -13,223 +14,206 @@
 
-    connect(this, SIGNAL(itemDoubleClicked(QTableWidgetItem*)), this,
-        SLOT(itemEditChanged(QTableWidgetItem*)));
+    connect(this, SIGNAL(itemDoubleClicked(QTableWidgetItem *)), this, SLOT(itemEditChanged(QTableWidgetItem *)));
 
-	RedoTable();
+    RedoTable();
     setWindowIcon(QIcon(tr(":/images/cur/icon-1-5.png")));
-    setWindowTitle(pDoc->m_base_title+":Table View");
+    setWindowTitle(pDoc->m_base_title + ":Table View");
 
     setVerticalScrollMode(QAbstractItemView::ScrollPerItem);
 
-	m_protect_edit = false;
+    m_protect_edit = false;
+
+    m_initialSize = settings.readSetting(SettingTag::depthmapViewSize, QSize(2000, 2000)).toSize();
 }
 
-tableView::~tableView()
-{
+TableView::~TableView() {
     // nothing todo
 }
 
-void tableView::RedoTable()
-{
-   clear();
-
-   if (pDoc->m_meta_graph->viewingProcessed()) {
-      const AttributeTable& table = pDoc->m_meta_graph->getAttributeTable();
-      int col = pDoc->m_meta_graph->getDisplayedAttribute();
-	  int i;
-
-      m_column_count = table.getNumColumns();
-	  setColumnCount(m_column_count+1);
-      connect(horizontalHeader(), SIGNAL(sectionClicked(int)), this, SLOT(colum_Sort(int)));
-	  verticalHeader()->hide();
-
-	  QTableWidgetItem *Item = new QTableWidgetItem("Ref Number");
-	  Item->setTextAlignment(Qt::AlignLeft);
-	  setHorizontalHeaderItem(0, Item);
-
-	  for (i = 0; i < m_column_count; i++) {
-         QTableWidgetItem *Item = new QTableWidgetItem(QString("%1").arg(table.getColumnName(i).c_str()));
-		 Item->setTextAlignment(Qt::AlignLeft);
-		 setHorizontalHeaderItem(i+1, Item);
-      }
-
-      m_row_count = table.getNumRows();
-	  setRowCount(m_row_count);
-      PrepareCache(m_curr_row);
-   }
+void TableView::RedoTable() {
+    clear();
+
+    if (pDoc->m_meta_graph->viewingProcessed()) {
+        const AttributeTable &table = pDoc->m_meta_graph->getAttributeTable();
+        int col = pDoc->m_meta_graph->getDisplayedAttribute();
+        int i;
+
+        m_column_count = table.getNumColumns();
+        setColumnCount(m_column_count + 1);
+        connect(horizontalHeader(), SIGNAL(sectionClicked(int)), this, SLOT(colum_Sort(int)));
+        verticalHeader()->hide();
+
+        QTableWidgetItem *Item = new QTableWidgetItem("Ref Number");
+        Item->setTextAlignment(Qt::AlignLeft);
+        setHorizontalHeaderItem(0, Item);
+
+        for (i = 0; i < m_column_count; i++) {
+            QTableWidgetItem *Item = new QTableWidgetItem(QString("%1").arg(table.getColumnName(i).c_str()));
+            Item->setTextAlignment(Qt::AlignLeft);
+            setHorizontalHeaderItem(i + 1, Item);
+        }
+
+        m_row_count = table.getNumRows();
+        setRowCount(m_row_count);
+        PrepareCache(m_curr_row);
+    }
 }
 
-void tableView::scrollContentsBy(int dx, int dy)
-{
-	if(!dy){
-		QTableWidget::scrollContentsBy(dx, 0);
-		return;
-	}
+void TableView::scrollContentsBy(int dx, int dy) {
+    if (!dy) {

```

Comment:
When touching this area - could we make this more explicit what condition we are checking?